app/models/metrics/slate_metrics_factory.py

```diff
@@ -0,0 +1,22 @@
+from typing import List, Dict
+
+from aws_xray_sdk.core import xray_recorder
+
+import app.config
+from app.models.metrics.metrics_model import MetricsModel
+from app.models.metrics.abstract_metrics_factory import AbstractMetricsFactory
+
+
+class SlateMetricsFactory(AbstractMetricsFactory):
+    _dynamodb_table: str = app.config.dynamodb['slate_metrics']['table']
```

Comment:
I'm trying to think how ORMs typically solve this. It's inconvenient to pass database credentials to the ORM on every query, so something has to live globally. Would it be better if we had a global instance of this object, but the credentials are passed in through an `__init__` function?

```python

class RecommendationMetricsFactory(AbstractMetricsFactory):
    _dynamodb_table: str = None
    _primary_key_name: str = None
    
    def __init__(dynamodb_table: str, primary_key_name: str):
        self._dynamodb_table = dynamodb_table
        self._primary_key_name = primary_key_name
    
    ...    


recommendationMetricsFactory = RecommendationMetricsFactory(
    dynamodb_table=app.config.dynamodb['recommendation_metrics']['table']
    primary_key_name=app.config.dynamodb['recommendation_metrics']['pk']
)
```

Reply:
So what I did in RecIt is pass all configuration information at object initialization (see https://github.com/Pocket/recit-ralph/blob/main/application.py#L74). This works very well b/c it encapsulates all the necessary state for the object in the object, rather than having it existing in the global environment.

Reply:
I like that pattern. It makes for code that's easier to test.

Would it be ok to apply it in a separate PR? This one has already gotten very big, that I'd like to apply that pattern consistently in this repo, rather than just for the classes that were touched here. I created a Jira ticket to track: https://getpocket.atlassian.net/browse/BACK-927

Reply:
sure, we can address it later.